api/events.proto

```diff
@@ -33,6 +34,8 @@ message Event {
   repeated string correlation_ids = 5 [(gogoproto.customname) = "CorrelationIDs", (validate.rules).repeated.items.string.max_len = 100];
   string origin = 6;
   map<string,bytes> context = 7;
+  // The event will be visible to a caller that has any of these rights.
+  Rights visibility = 8;
```

Comment:
I think `visibility` can be a bit unclear in this case - how about e.g. `RequiredRights`?

Reply:
Yeah I was struggling with that too and changed it a couple of times myself.

- `Rights rights` - what rights?
- `Rights required_rights` - rights required for what?
- `Rights required_rights_for_read` - that's long
- `Rights visibility` - not perfect but not terrible either